packages/schematics/angular/migrations/update-10/update-tslint.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonAstObject, JsonValue } from '@angular-devkit/core';
+import { Rule } from '@angular-devkit/schematics';
+import { addPackageJsonDependency, getPackageJsonDependency } from '../../utility/dependencies';
+import { findPropertyInAstObject, insertPropertyInAstObjectInOrder, removePropertyInAstObject } from '../../utility/json-utils';
+import { readJsonFileAsAstObject } from '../update-9/utils';
+
+export const TSLINT_VERSION = '~6.1.0';
+const TSLINT_CONFIG_PATH = '/tslint.json';
+
+const RULES_TO_DELETE: string[] = [
+  'no-use-before-declare',
+  'no-unused-variable',
+];
+
+const RULES_TO_ADD: Record<string, JsonValue> = {
+  align: {
+    options: ['parameters', 'statements'],
+  },
+  'arrow-return-shorthand': true,
+  curly: true,
+  eofline: true,
+  'import-spacing': true,
+  indent: {
+    options: ['spaces'],
+  },
+  semicolon: { options: ['always'] },
+  'space-before-function-paren': {
+    options: {
+      anonymous: 'never',
+      asyncArrow: 'always',
+      constructor: 'never',
+      method: 'never',
+      named: 'never',
+    },
+  },
+  'typedef-whitespace': {
+    options: [
+      {
+        'call-signature': 'nospace',
+        'index-signature': 'nospace',
+        parameter: 'nospace',
+        'property-declaration': 'nospace',
+        'variable-declaration': 'nospace',
+      },
+      {
+        'call-signature': 'onespace',
+        'index-signature': 'onespace',
+        parameter: 'onespace',
+        'property-declaration': 'onespace',
+        'variable-declaration': 'onespace',
+      },
+    ],
+  },
+  whitespace: {
+    options: [
+      'check-branch',
+      'check-decl',
+      'check-operator',
+      'check-separator',
+      'check-type',
+      'check-typecast',
+    ],
+  },
+};
+
+export default function (): Rule {
+  return (tree, context) => {
+    const logger = context.logger;
+
+    // Update tslint dependency
+    const current = getPackageJsonDependency(tree, 'tslint');
+
+    if (!current) {
+      logger.warn('"tslint" in not a dependency of this workspace.');
+
+      return;
+    }
+
+    if (current.version !== TSLINT_VERSION) {
+      addPackageJsonDependency(tree, {
+        type: current.type,
+        name: 'tslint',
+        version: TSLINT_VERSION,
+        overwrite: true,
+      });
+    }
+
+    // Update tslint config.
+    const tslintJsonAst = readJsonFileAsAstObject(tree, TSLINT_CONFIG_PATH);
```

Comment:
tslint config files can also be JS or YAML in addition to JSON.  Maybe check, warn, and skip if one of those is found?

There can also be additional configs in subdirectories.  tslint has a discovery mechanism wherein the _closest_ tslint configuration per linted file is used for that file unless a configuration file is explicitly specified upon execution.